### PR TITLE
Add `--tool` flag to specify how to build the crate before flashing.

### PR DESCRIPTION
**Pull request by MabezDev**
Cool project! Has been very useful in removing hacky scripts that invoke esptool.

This PR adds functionality to choose how to build the project, instead of just relying on xargo. I'm attempting to rewrite the xtensa-quickstart repo without needed any other tools to build the code. If it goes well I will be recommending this tool for sure :)

### CODE REVIEW

File: cargo-espflash/README.md [hunk 1]
```diff
@@ -2,7 +2,13 @@
 
 _ESP8266_ and _ESP32_ cross-compiler and serial flasher cargo subcommand.
 
-Currently, `cargo-espflash` requires that you have [xargo](https://github.com/japaric/xargo) installed on your system.
+<!-- Currently, `cargo-espflash` requires that you have [xargo](https://github.com/japaric/xargo) installed on your system. -->
```
Comment (jessebraham) [inline cargo-espflash/README.md:5]: This comment can probably be removed.

File: cargo-espflash/README.md [hunk 2]
```diff
@@ -2,7 +2,13 @@
...
+
+To build the project before flashing, `cargo-espflash` has a few options, specified with the `--tool TOOL` flag.
+
+ - `--tool cargo`, build using the `build-std` unstable cargo feature. This is the default option.
+ - `--tool xargo`, build using `xargo`. Requires [xargo](https://github.com/japaric/xargo) installed on your system.
+ - `--tool xbuild`, build using `cargo xbuild`. Requires [cargo xbuild](https://github.com/rust-osdev/cargo-xbuild) installed on your system.
 
 ## Usage
```
Comment (jessebraham) [inline cargo-espflash/README.md:13]: Could you please add `[--tool TOOL]` to the usage example below?

File: cargo-espflash/src/main.rs [hunk 3]
```diff
@@ -68,12 +83,13 @@ struct AppArgs {
...
     example: Option<String>,
     chip: Option<String>,
+    build_tool: Option<String>,
     serial: Option<String>,
 }
 
 fn usage() -> Result<(), MainError> {
     let mut usage = String::from("Usage: cargo espflash ");
-    usage += "[--ram] [--release] [--example EXAMPLE] ";
+    usage += "[--ram] [--release] [--example EXAMPLE] [--build-tool {{cargo,xargo,xbuild}}]";
```
Comment (jessebraham) [inline cargo-espflash/src/main.rs:92]: I believe this should be `[--tool {{cargo,xargo,xbuild}}]`.

**Review by jessebraham — COMMENTED** [review]
Thanks for this! I wanted to include this in the original version but ran out of steam 😄 Just a couple of small nitpicks but otherwise looks pretty good to me.

**Comment by MabezDev** [comment]
I think I've addressed all your comments, apologies for the delay I've been working through some `build-std` issues (around compiler builtins), acting on that I've chosen to promote `cargo xbuild` to the default build tool until those issues are resolved.

**Comment by jessebraham** [comment]
LGTM, thanks! I can't merge in this repo (otherwise I would) so @icewind1991 will have to take care of that.

**Pull request merged**